sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2907,6 +2907,18 @@ object SQLConf {
       .checkValue(bit => bit >= 10 && bit <= 30, "The bit value must be in [10, 30].")
       .createWithDefault(16)
 
+  val ADAPTIVE_PARTIAL_AGGREGATION_THRESHOLD =
+    buildConf("spark.sql.aggregate.adaptivePartialAggregationThreshold")
+      .internal()
+      .doc("Minimum number of processed rows before partial aggregation can be skipped. " +
+        "By setting this value to 0 adaptive partial aggregation can be disabled.")
+      .version("3.3.0")
+      .intConf
+      .checkValue(threshold => threshold >= 0 && threshold < (1 << 16),
+        "The threshold value must be bigger than or equal to 0 and less than " +
+          s"1 << ${FAST_HASH_AGGREGATE_MAX_ROWS_CAPACITY_BIT.key}.")
+      .createWithDefault(10000)
```

Comment:
Just curious, any reason that the default value of adaptivePartialAggregationThreshold is 10,000? [In Trino the default is 100,000](https://github.com/trinodb/trino/blob/ab7bc3fa534110893b96232b6b1c6d36286dcbb6/core/trino-main/src/main/java/io/trino/sql/planner/OptimizerConfig.java#L86)